### PR TITLE
Merge 1.5 Dev branch into master

### PR DESCRIPTION
**Pull request by tleyden**
This PR merges the 1.5 Dev branch into master

### CODE REVIEW

**Comment by tleyden** [comment]
Integration tests in progress:

http://uberjenkins.sc.couchbase.com/view/Build/job/build-sync-gateway/271/
http://uberjenkins.sc.couchbase.com/view/Build/job/build-sync-gateway/272/

File: base/bucket.go [hunk 1]
```diff
@@ -16,13 +16,14 @@ import (
 	"strconv"
 	"strings"
 
+	"time"
```
Comment (adamcfraser) [inline base/bucket.go]: Whitespace between strings and time.

**Review by adamcfraser — COMMENTED** [review]
Minor import formatting comment.

File: base/bucket.go (hunk 1 above)
Comment (tleyden) [reply base/bucket.go]: Fixed

**Comment by tleyden** [comment]
Looking at commit https://github.com/couchbase/sync_gateway/commit/cd341535e6187b6c295b23093a4ae76c12d63449, a lot of commits got collapsed into a single commit .. I think next time we do this we shouldn't do a "squash".  (in a sense it's a double-squash, since all of those were squashed once during their PR's)

**Pull request merged**